### PR TITLE
03 use redux in react components 

### PR DESCRIPTION
**Pull request by FaranoshAmini**
In this pull request, I have implemented:

 - Keep state immutables
 - Initialize the array with a few books
 - Connect the app with Redux Store
 - Use useDispatch and useSelector
 - Add functionality to add a new book
 - Add functionality to remove the book
 - Check the status of categories

### CODE REVIEW

File: src/components/BookForm.js [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/books';
 
 function BookForm() {
+  const dispatch = useDispatch();
+
+  const handleAddBook = (e) => {
+    e.preventDefault();
+    const title = e.target.title.value;
+    const author = e.target.author.value;
```
Comment (ShahierNashaat) [inline src/components/BookForm.js]: - [ ] Kindly notice that you use a local state with `title` and `author` as per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/use_redux_in_react_components.md#:~:text=Set%20the%20data,to%20an%20object.).

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @FaranoshAmini, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Wes-Isaac — APPROVED** [review]
Hi @FaranoshAmini, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Great PR description 👍.
- Professional README 💯.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by FaranoshAmini** [comment]
@Wes-Isaac Thanks for your kind review.

**Pull request merged**